### ISSUE TITLE
Speech Note 4.7.0 Beta 4

### ISSUE THREAD

**Issue by mkiol** [mkiol/dsnote#179]
If you want to test the upcoming release, **Speech Note 4.7.0 Beta 3** is available in "flathub-beta" repository. 

This version is perfectly usable, but may contain more bugs.

To enable "flathub-beta" in your system follow [this instruction](https://discourse.flathub.org/t/how-to-use-flathub-beta/2111) or simply do the following:

```
flatpak remote-add --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
```

## Release Highlights

- Vulkan GPU acceleration for _WhisperCpp_

## Changes between `4.6.1` and `4.7.0 Beta 4`

- General
  - _Rules_ for text transformations that can be applied after Speech to Text or before Text to Speech. With _Rules_, you can easily and flexibly correct errors in decoded text or correct mispronounced words.
  - New modes for inserting text at the cursor position or replacing the current note. To insert text at the cursor position rather than at the end of the note, change _Text appending mode_ option to _Add at the cursor position_ in the settings. When the _Replace an existing note_ option is set, whenever new text is added, it will replace the existing note.
  - DBus API for integration with external applications
- User Interface:
  - Status indication in the system tray icon. When using the system tray icon, statuses such as processing, listening, etc. are presented with an animated tray icon.
  - Models grouped by type in model browser. To improve usability, instead of a list containing models of all types, models are grouped by type in separate tabs.
  - New _General_ and _Advanced_ tabs in _Settings_
  - Command line options for printing available or active model IDs. Use _--print-available-models_ or _--print-active-model_ to list all available models or the currently active model.
  - Command line option to print the current status of the application. Use _--print-state_ to see the current state. This option can be useful when integrating with external programs or widgets on the desktop.
- Speech to Text:
  - Support for Vulkan GPU acceleration in _WhisperCpp_. Vulkan acceleration enables much faster STT decoding with Intel, AMD or NVIDIA graphics cards. With Vulkan, decoding is quicker than with OpenVINO, OpenCL and ROCm, but still may be slightly slower compared to CUDA. The biggest advantage of Vulkan is that you can use it without installing any GPU acceleration add-ons. Vulkan is not enabled by default for integrated GPU. To test it on iGPU, enable _Use Vulkan iGPU_ in the settings (Other->Hardware acceleration options).
  - New _Whisper Large Turbo_ model for both _WhisperCpp_ and _FasterWhisper_. _Turbo_ is a finetuned version of a pruned _Whisper Large-v3_. It's the exact same model, except that the number of decoding layers have reduced. As a result, the model is way faster, at the expense of a minor quality degradation. _Turbo_ model does not have the ability to translate into English, as does the regular Large model.
  - Simplified engine configuration options. Instead of multiple options, you can now select a Profile, which allows you to change the engine's processing parameters. There are three profiles to choose from: _Best Performance_, _Best Quality_ and _Custom_.
  - Echo mode. After processing, the decoded text will be immediately read out using the currently set Text to Speech model.
- Text to Speech:
  - New _Piper_ voice for Latvian
  - New _WhisperSpeech Small_ model for: English, Italian, German, French, Spanish, Dutch and Portuguese
- Translator:
  - New models: English to Finnish, English to Turkish, English to Swedish, Swedish to English, English to Slovak, English to Indonesian, English to Romanian, English to Greek, Chinese to English
  - Updated models: English to Catalan, English to Russian, English to Ukrainian, English to Czech
- Accessibility:
  - Option to scan special key strokes when setting keyboard shortcuts (X11 only). If you want to use special keys as shortcuts (so-called "multimedia keys"), instead of typing their names, you can automatically set the key by pressing it.
  - Keyboard shortcuts enabled for several user interface elements. Elements such as menu items or buttons can be controlled using the keyboard shortcuts. Examples: Switch to Notepad (Ctrl+N), Switch to Translator (Ctrl+T), Open Languages (Ctrl+L), Read (Ctrl+Alt+Shift+R), Listen (Ctrl+Alt+Shift+L), Stop (Ctrl+Alt+Shift+S), Cancel (Ctrl+Alt+Shift+C), Pause (Ctrl+Alt+Shift+P) and more...
  - New _Actions_ and global keyboard shotcut to force translation of text in STT: _start-listening-translate_, _start-listening-translate-active-window_, _start-listening-translate-clipboard_. The decoded text is always translated into English when the "translate" action is triggered. This only works when using Whisper models.
  - New _Action_ to read text from the command-line option: _start-reading-text_. To pass text, use the _--text_ option in the command-line interface.
- Flatpak:
  - whisper.cpp update to version 1.7.1
  - PyTorch update to version 2.5.1
  - ROCm update to version 6.2.2 (AMD add-on)
  - cuDNN update to version 9.5.1 (NVIDIA add-on)

Make sure to update the add-on to version `1.3.0` if you are using it.

**Comment by mkiol** [comment]
The new official version has been released. A video presentation of all the new features in 4.7.0 can be found [here](https://www.youtube.com/watch?v=cEht4Fts6Bo).
